parse/apk_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package parse_test
+
+import (
+	"strings"
+	"testing"
+
+	. "github.com/sonatype-nexus-community/ahab/parse"
+	"github.com/sonatype-nexus-community/nancy/types"
+	"github.com/stretchr/testify/assert"
+)
+
+// generate CLI package list via:
+// # apk info -vv | sort
+var apkShowList = `WARNING: Ignoring APKINDEX.00740ba1.tar.gz: No such file or directory
+WARNING: Ignoring APKINDEX.d8b2a6f4.tar.gz: No such file or directory
+alpine-baselayout-3.1.2-r0 - Alpine base dir structure and init scripts
+alpine-keys-2.1-r2 - Public keys for Alpine Linux packages
+apk-tools-2.10.4-r2 - Alpine Package Keeper - package manager for alpine
+busybox-1.30.1-r2 - Size optimized toolbox of many common UNIX utilities
+ca-certificates-cacert-20190108-r0 - Mozilla bundled certificates
```

Comment:
There may be a boog lurking here. The `ca-certificates-cacert-20190108-r0` case produced an empty version string, where I think it should produce `20180409-r0`

e.g.

    // ca-certificates 20180409
	assert.Equal(t, types.Projects{"ca-certificates", "20180409-r0"}, result.Projects[4])

Not sure it helps, but here's the regex I was using when I started to implement APK. Mine did not use `apk info -vv`, but only `apk info -v`, so less verbose. I was also removing the "-rN" suffix because I was not sure how ossindex would handle versions with the "-rN" suffix. Different inputs, so not sure if it helps much, but just in case...

```
func doApkParse(pkg string) (parsedProject types.Projects) {
	pkg = strings.TrimSpace(pkg)

	re, err := regexp.Compile(`^((.*)-([^a-zA-Z].*)-.*)`)
	if err != nil {
		fmt.Println(err)
	}
	newSlice := re.FindStringSubmatch(pkg)
	if newSlice != nil {
		parsedProject.Name = newSlice[2]
		parsedProject.Version = newSlice[3]
	} else {
		// yikes, nothing we recognize.
		fmt.Printf("filaure parseing namve,version for package: %s\n", pkg)
	}
	return
}
```

Reply:
In: 6e673b8